### PR TITLE
Android non-biometry support + ios localization

### PR DESCRIPTION
**Pull request by killalad**

### CODE REVIEW

File: android/src/main/kotlin/design/codeux/biometric_storage/BiometricStoragePlugin.kt [hunk 1]
```diff
@@ -254,13 +254,20 @@ class BiometricStoragePlugin : FlutterPlugin, ActivityAware, MethodCallHandler {
...
+                .setTitle(promptInfo.title)
+                .setSubtitle(promptInfo.subtitle)
+                .setDescription(promptInfo.description)
+                .setConfirmationRequired(promptInfo.confirmationRequired)
+
+        if (!options.biometryOnly) {
+            promptBuilder.setDeviceCredentialAllowed(true)
+        } else {
+            promptBuilder.setNegativeButtonText(promptInfo.negativeButton)
+        }
```
Comment (hpoul) [inline android/src/main/kotlin/design/codeux/biometric_storage/BiometricStoragePlugin.kt:268]: I think `setDeviceCredentialAllowed` is deprecated, and maybe we should make the intent more explicit:

```kotlin
if (options.biometryOnly) {
  promptBuilder
    .setAllowedAuthenticators(BIOMETRIC_WEAK | BIOMETRIC_STRONG)
    .setNegativeButtonText(promptInfo.negativeButton)
} else {
  promptBuilder
    .setAllowedAuthenticators(DEVICE_CREDENTIAL | BIOMETRIC_WEAK | BIOMETRIC_STRONG)
}
```

this way it's obvious that the flag simply adds `DEVICE_CREDENTIAL`

File: lib/src/biometric_storage.dart [hunk 2]
```diff
@@ -120,6 +126,36 @@ class AndroidPromptInfo {
       };
 }
 
+/// iOS specific configuration of the prompt displayed for biometry.
+class IOSPromptInfo {
```
Comment (hpoul) [inline lib/src/biometric_storage.dart:130]: I think i would prefer having it called `IosPromptInfo` in dart - i think that's the [preferred capitalisation convention within flutter] 🤔️(https://github.com/flutter/plugins/blob/09ec5192f3ce31622631458ba9655cc257a07951/packages/device_info/device_info_platform_interface/lib/model/ios_device_info.dart#L8). (also like `Uri`, etc. - although i'm not sure why its `Platform.iOS` 😅️

File: android/src/main/kotlin/design/codeux/biometric_storage/BiometricStoragePlugin.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (killalad) [reply android/src/main/kotlin/design/codeux/biometric_storage/BiometricStoragePlugin.kt]: yeah you're right

File: lib/src/biometric_storage.dart [hunk 3]
```diff
@@ -86,10 +87,15 @@ class StorageFileInitOptions {
   /// will simply be save encrypted. (default: true)
   final bool authenticationRequired;
 
+  /// Only makes difference on Android, where if set true, you can't use
+  /// PIN/pattern/password to get the file.
+  final bool biometryOnly;
```
[the remark this answers is not in the document]
Comment (killalad) [reply lib/src/biometric_storage.dart]: sure that makes sense

File: lib/src/biometric_storage.dart (hunk 2 above)
Comment (killalad) [reply lib/src/biometric_storage.dart]: Okay changed 😄 , seems to me that dart is full of these inconsistencies, but which language isn't 😄

**Comment by hpoul** [comment]
thanks, looks good to me 👍️

i've just renamed `androidBiometryOnly` to `androidBiometricOnly` and rebased on master.

*merge*

**Comment by hpoul** [comment]
fyi, i've encountered two problems

1.  I forgot that iOS and MacOS uses the same implementation, so the MacOS version crashed because it was also looking for `iosPromptInfo`. I'm now simply passing the same prompt info to MacOS as well. https://github.com/authpass/biometric_storage/commit/acdf6eff8a3efbca202696f8435f0ea250328577
2. To keep the same behavior i've changed the default for `androidMetricOnly` to `true`. I still don't quite understand if it should be completely customizable (ie. also let users change weak/strong) but anyway. at least this way the old behavior stays the same.

**Pull request merged**